### PR TITLE
Create modules in pkg/logs and move internal pkg/logs dependencies

### PR DESCRIPTION
**Pull request by liustanley**
### What does this PR do?

This PR creates the following modules in `pkg/logs`:
- `pkg/logs/auditor`
- `pkg/logs/client`
- `pkg/logs/diagnostic`
- `pkg/logs/logstatus`
- `pkg/logs/message`
- `pkg/logs/metrics`
- `pkg/logs/pipeline`
- `pkg/logs/processor`
- `pkg/logs/sender`
- `pkg/logs/sources`
- `pkg/logs/util/testutils`
- `pkg/util/startstop` (dependency of `pkg/logs/client` and `pkg/logs/pipeline`)

In doing so, we had to move the following internal packages out of `pkg/logs/internal` in order to convert the other `pkg/logs` packages into modules:
- `pkg/logs/internal/processor` -> `pkg/logs/processor`
- `pkg/logs/internal/status` -> `pkg/logs/status/utils`
- `pkg/logs/internal/util/testutils` -> `pkg/logs/util/testutils`

### Motivation

We need to convert these packages in `pkg/logs` into modules in order to use the logs agent in the OTel Collector Datadog Exporter as described in the following [RFC](https://docs.google.com/document/d/1uKWa9rMNm0U-euEzEJ69KW7Id5u1E4GfYHyzC1SzYoI/edit#heading=h.wnkz5u7vx1fb).

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/logs/logstatus/info.go [hunk 1]
```diff
@@ -4,7 +4,7 @@
 // Copyright 2016-present Datadog, Inc.
 
 //nolint:revive // TODO(AML) Fix revive linter
-package status
+package logstatus
```
Comment (gh123man) [inline pkg/logs/logstatus/info.go:7]: Can we put this code in or in a sub-package of `pkg/logs/status`? 
It's unclear from a quick glance why we need a `pkg/logs/status` and `pkg/logs/logstatus` as they both sound like they do the same thing.

Comment (liustanley) [reply pkg/logs/logstatus/info.go:7]: Sure, do you want to keep the name `logstatus` for the package? I can move it to `pkg/logs/status/logstatus` if that works.

**Review by gh123man — APPROVED** [review]

**Review by spencergilbert — APPROVED** [review]

**Review by dustmop — APPROVED** [review]
LGTM

**Review by GustavoCaso — CHANGES_REQUESTED** [review]
> Would it be too complex to create a single pkg/logs module ?

@liustanley I think having that much granularity in the number of modules we create could create maintainability issues in the future. 

Is there any reason whey we do not create a single `pkg/logs` module?

**Review by clarkb7 — APPROVED** [review]

**Comment by liustanley** [comment]
@GustavoCaso @pgimalac We can't create a single `pkg/logs` module because of other coupled dependencies throughout datadog-agent. For example, `pkg/logs/tailers` depends on `pkg/util/docker` and `pkg/util/containers` which both aren't modules and have many other dependencies which also aren't modules. Because of cases such as this, we chose to only convert the packages in `pkg/logs` that are needed to export the logs pipeline for our use in the Collector.

**Review by GustavoCaso — CHANGES_REQUESTED** [review]
> We can't create a single pkg/logs module because of other coupled dependencies throughout datadog-agent. For example, pkg/logs/tailers depends on pkg/util/docker and pkg/util/containers which both aren't modules and have many other dependencies which also aren't modules. Because of cases such as this, we chose to only convert the packages in pkg/logs that are needed to export the logs pipeline for our use in the Collector.

I understand that. I haven't thought much of alternatives solutions to that problem yet, but I think we would benefit greatly if we could change how we approach the creation of modules. The dependency graph in the agent repo is far from ideal, and maybe we have a chance of cleaning it up as we create modules for Otel. Eitherway I believe that is a conversation for outside of the scope of this PR 😄 

The request changes comment is not about the creation of the different packages in the `pkg/logs` but because of pattern described in this comment below

I'll reach to you in Slack so we can discuss possible alternatives to the `pkg/logs/status/statusinterface/status.go` package

**Review by GustavoCaso — APPROVED** [review]
LGTM. Thank you so much for taking the time to address my comments regarding the status provider interface 😄 🎉

**Pull request merged**